### PR TITLE
Module to write sbatch scripts

### PR DESCRIPTION
**Pull request by adrosenbaum**
Improved mutacc_auto/files/sbatch.py

### CODE REVIEW

File: mutacc_auto/files/sbatch.py [hunk 1]
```diff
@@ -25,59 +23,81 @@ def get_timestamp():
...
+    ('A', ACCOUNT),
+    ('n', NODES),
+    ('t', TIME),
+    ('J', JOBNAME),
+    ('qos', PRIORITY),
+    ('mail-type', MAIL_FAIL),
+    ('mail-type', MAIL_END)
 )
 
+ACTIVATE_ENVIRONMENT = "source activate"
```
Comment (henrikstranneheim) [inline mutacc_auto/files/sbatch.py]: with more recent versions of conda this is now "conda activate"

File: mutacc_auto/files/sbatch.py [hunk 2]
```diff
@@ -25,59 +23,81 @@ def get_timestamp():
...
+
+STDERR_SUFFIX = "err"
+STDOUT_SUFFIX = "out"
+
+NEWLINE = "\n"
```
Comment (henrikstranneheim) [inline mutacc_auto/files/sbatch.py]: I think you should add this in a Constants.py and import it here

File: mutacc_auto/files/sbatch.py [hunk 3]
```diff
@@ -25,59 +23,81 @@ def get_timestamp():
...
-        header += f"{HEADER_PREFIX} -o {stdout_file}\n"
-        header += f"{HEADER_PREFIX} --mail-user={email}\n"
+            if len(option[0]) == 1:
+                header += f"{HEADER_PREFIX} -{option[0]} {option[1]}{NEWLINE}"
+            else:
+                header += f"{HEADER_PREFIX} --{option[0]}={option[1]}{NEWLINE}"
+
+        #make log files
+        log_directory = Path(log_directory)
+        if not log_directory.is_dir():
```
Comment (henrikstranneheim) [inline mutacc_auto/files/sbatch.py]: Can't this check be handled by Click?

File: tests/files/test_sbatch.py [hunk 4]
```diff
@@ -10,27 +10,33 @@ def test_SbatchScript(tmpdir):
...
 
     sbatch_path = sbatch_script.path
 
     assert Path(sbatch_path).exists()
 
+    with pytest.raises(FileNotFoundError) as error:
+
+        sbatch_script = SbatchScript(tmp_dir,
+                                     'mutacc_env',
+                                     '/fds9Xs/fdsuerw')
```
Comment (henrikstranneheim) [inline tests/files/test_sbatch.py]: '/fds9Xs/fdsuerw'
This looks very hard coded! What is it? A temp dir again?

**Review by henrikstranneheim — COMMENTED** [review]
100% in coverage 🎉 . Nice!

File: mutacc_auto/files/sbatch.py (hunk 1 above)
Comment (adrosenbaum) [reply mutacc_auto/files/sbatch.py]: ok, I'll give some support for 'conda activate' as well!

File: mutacc_auto/files/sbatch.py (hunk 2 above)
Comment (adrosenbaum) [reply mutacc_auto/files/sbatch.py]: will do!

File: mutacc_auto/files/sbatch.py (hunk 3 above)
Comment (adrosenbaum) [reply mutacc_auto/files/sbatch.py]: Yes you are right! this is unnecessary code. will remove.

File: tests/files/test_sbatch.py (hunk 4 above)
Comment (adrosenbaum) [reply tests/files/test_sbatch.py]: This is just to test the function with a nonsense path. will change.

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**